lib/connextclient/errors.ts

```diff
@@ -0,0 +1,77 @@
+import errorCodesPrefix from '../constants/errorCodesPrefix';
+import { ClientStatus } from '../swaps/SwapClient';
+
+const codesPrefix = errorCodesPrefix.CONNEXT;
+const errorCodes = {
+  CONNEXT_IS_DISABLED: codesPrefix.concat('.1'),
+  INSUFFICIENT_BALANCE: codesPrefix.concat('.2'),
+  TIMEOUT: codesPrefix.concat('.3'),
+  INVALID: codesPrefix.concat('.4'),
+  SERVER_ERROR: codesPrefix.concat('.5'),
+  UNEXPECTED: codesPrefix.concat('.6'),
+  TOKEN_ADDRESS_NOT_FOUND: codesPrefix.concat('.7'),
+  INVALID_TOKEN_PAYMENT_RESPONSE: codesPrefix.concat('.8'),
+  CONNEXT_IS_UNAVAILABLE: codesPrefix.concat('.9'),
+  CONNEXT_HAS_NO_ACTIVE_CHANNELS: codesPrefix.concat('.10'),
+  CONNEXT_CLIENT_NOT_INITIATED: codesPrefix.concat('.11'),
+  CONNEXT_WALLET_NOT_INITIATED: codesPrefix.concat('.12'),
+};
+
+const errors = {
+  CONNEXT_IS_DISABLED: {
+    message: 'connext is disabled',
+    code: errorCodes.CONNEXT_IS_DISABLED,
+  },
+  CONNEXT_IS_UNAVAILABLE: (status: ClientStatus) => ({
+    message: `connext is ${ClientStatus[status]}`,
+    code: errorCodes.CONNEXT_IS_UNAVAILABLE,
+  }),
+  CONNEXT_HAS_NO_ACTIVE_CHANNELS: () => ({
+    message: 'connext has no active channels',
+    code: errorCodes.CONNEXT_HAS_NO_ACTIVE_CHANNELS,
+  }),
+  CONNEXT_CLIENT_NOT_INITIATED: {
+    message: 'connext client is not initiated',
+    code: errorCodes.CONNEXT_CLIENT_NOT_INITIATED,
+  },
+  CONNEXT_WALLET_NOT_INITIATED: {
+    message: 'connext wallet is not initiated',
+    code: errorCodes.CONNEXT_WALLET_NOT_INITIATED,
+  },
+  INSUFFICIENT_BALANCE: {
+    message: 'insufficient balance to perform request',
+    code: errorCodes.INSUFFICIENT_BALANCE,
+  },
+  TIMEOUT: {
+    message: 'request timed out',
+    code: errorCodes.TIMEOUT,
+  },
+  /**
+   * For HTTP 409 responses from Connext. For token payments 409 means "the address
+   * or the amount is invalid, or there is no path to the target, or the identifier
+   * is already in use for a different payment."
+   */
+  INVALID: (message: string) => ({
```

Comment:
It seems like most (or all?) of the errors from this point down aren't being used, but are rather carryovers from the Raiden REST HTTP error codes. No need to have them here, errors should be Connext specific and only included if they're actually being used.

Reply:
Correct, however they will be useful for using the new REST API client that Im working on